### PR TITLE
Build MTE-280 [v109] Run Selective UITests

### PR DESCRIPTION
**Pull request by isabelrios**
The idea of this PR is to run UI tests only when there are app, test changes or changes in files that could affect the app or tests, and otherwise just build and run the unit tests.

See the [issue](https://mozilla-hub.atlassian.net/browse/MTE-280) in jira and the [doc](https://docs.google.com/document/d/1acEswsZzuzPA7Y_H2NpdQiKeIjcVdKZKn_kq1Rqn4iI/edit?usp=sharing) created for the task.

### CODE REVIEW

File: Shared/af.lproj/AuthenticationManager.strings
Comment (isabelrios) [inline Shared/af.lproj/AuthenticationManager.strings:1]: Change to show that PR does not run the test if a file not related to the app or tests (as decided in a list) is modified
https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/903644ff-c690-48ae-b3fa-37b35de3211e
UI tests do not run based on the ENV VAR

File: bitrise.yml
Comment (isabelrios) [inline bitrise.yml:16]: ⚠️ Tem changes to set back

File: bitrise.yml
Comment (isabelrios) [inline bitrise.yml:25]: We can remove this from here, added to see if the scripts was working in the config step. We can't save and share the env variable value from different stages and workflows that's why we would need to run the workflow within each test workflow and not here

File: bitrise.yml
Comment (isabelrios) [inline bitrise.yml:1449]: Not intended change, will undo this

File: bitrise.yml
Comment (isabelrios) [inline bitrise.yml:1452]: Same as above

File: bitrise.yml
Comment (isabelrios) [inline bitrise.yml:1464]: same as above

File: bitrise.yml
Comment (clarmso) [inline bitrise.yml:634]: 🤔 Could we use CI's syntax to handle when the tests are run? I was thinking about this syntax from Github Actions.
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#using-filters

Sorry I am not proficient in Bitrise's syntax yet. 🙇🏼‍♀️  Is it possible to use something like this? We can use the regular expression of the files' paths to denote the changes that UI tests are not needed. 💡 
https://devcenter.bitrise.io/en/builds/configuring-build-settings/selective-builds.html

File: bitrise.yml
Comment (AaronMT) [inline bitrise.yml:610]: Just double checking if this decision model always intended for `main` branch as you target here?

File: bitrise.yml
Comment (AaronMT) [inline bitrise.yml:634]: Just double checking if you need to escape the `/` characters `\/`, tested the entire query on https://regexr.com with `Docs/` and `Shared/` and it errored.

Will this run if a commit contains both code where we want UI tests to run and code where we don't need UI tests to run? E.g, I commit a change to `Docs` as well as core browser code.

File: bitrise.yml
Comment (isabelrios) [reply bitrise.yml:610]: So far, yes. Only checking against main, but once we have this running, we could add more branches

File: bitrise.yml
Comment (isabelrios) [reply bitrise.yml:634]: mm locally it works for me without scaping those, but I will double check to be sure it also works on CI.
Yes, I have added the logic to cover these scenarios:
No changes: UI tests do not run

> Changes in no "app" files AND "app" files: UI tests run
> Changes in "app" files: UI tests run
> Changes in no "app" files: UI tests do not run

Comment (isabelrios) [reply bitrise.yml]: This is a script step on Bitrise, this means is like creating a shell script and that's the syntaxt. There is not any special syntax on Bitrise, in general it is a .yml fil, you have to follow that syntax, and then there are different type of steps, each require some options if they are pre-defined steps on bitrise or shell scripting if they are script steps.
This is the [bitrise documentation](https://devcenter.bitrise.io/en/steps-and-workflows/introduction-to-steps.html) about steps

> Sorry I am not proficient in Bitrise's syntax yet. 🙇🏼‍♀️ Is it possible to use something like this? We can use the regular expression of the files' paths to denote the changes that UI tests are not needed. 💡
> https://devcenter.bitrise.io/en/builds/configuring-build-settings/selective-builds.html

We can't use the default selective builds option Bitrise has, since it allows only the path to one file. The idea of that option is build or not apps depending on some changes if you have more than one app in the same repo for example

**Comment by isabelrios** [comment]
I would like to give this a try for some time and see if we benefit from this implementation, if there are issues with it and if there might be opportunities to make it better.
We can land this next Monday Nov. 28th and carefully monitor the builds during that week. 

Sharing a few examples:
[Build](https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/903644ff-c690-48ae-b3fa-37b35de3211e) not running UI Tests because only one locale file changed
[Build](https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/cd3203e4-b124-4345-b6f4-d644315d74ef) running regularly as bitrise file changed

Does this sound good @nbhasin2 ?

**Review by AaronMT — APPROVED** [review]

**Comment by nbhasin2** [comment]
Giving a late LGTM after merge

**Pull request merged**
